src/components/DropdownInput/DropdownInput.tsx

```diff
@@ -1,34 +1,35 @@
-import Link from 'next/link';
-import { ReactElement, PropsWithChildren, ChangeEvent } from 'react';
+import { ReactElement, PropsWithChildren, ChangeEvent, useState } from 'react';
+// eslint-disable-next-line import/no-cycle
+import { russianRegion } from '../LocationSelector/Russia/RussianCitiesSelect';
+import { DropdownCities } from '../DropdownCities';
 
 type DropdownVariant = 'fullWidth' | 'halfWidth' | 'width75' | 'width25';
 
 export interface DropdownInputProps extends PropsWithChildren {
   variant: DropdownVariant;
-  onChange?: (event: ChangeEvent<HTMLInputElement>) => void;
   placeholder: string;
   searchItem: string;
-  searchResultsHref: string;
-  searchResultsText: string;
-  inputValue: string;
 }
 
 export const DropdownInput = ({
-  onChange,
   variant,
   placeholder,
   children,
   searchItem,
-  searchResultsHref,
-  searchResultsText,
-  inputValue,
 }: DropdownInputProps): ReactElement => {
   const classes: Record<DropdownVariant, string> = {
     fullWidth: 'w-full',
     halfWidth: 'w-1/2',
     width75: 'w-9/12',
     width25: 'w-3/12',
   };
+  const [search, setSearch]: [string, (search: string) => void] = useState('');
```

Comment:
I am not sure if you need the `[string, (search: string) => void` type, it should be picked up autimatically by TS